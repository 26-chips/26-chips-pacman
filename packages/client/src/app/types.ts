export type User = {
  id: number;
  login: string;
  display_name: string;
  first_name: string;
  second_name: string;
  avatar: string;
  email: string;
  phone: string;
};

export type SigninData = {
  login: string;
  password: string;
};

export type SignupData = {
  email: string;
  first_name: string;
  second_name: string;
  phone: string;
  login: string;
  password: string;
  repeat_password: string;
};

export type ProfileData = Pick<User, 'login' | 'first_name' | 'second_name' | 'email' | 'phone'> & { display_name: string };

export type PasswordData = {
  oldPassword: string;
  newPassword: string;
  repeatPassword: string;
};
