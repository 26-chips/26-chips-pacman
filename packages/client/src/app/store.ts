import { configureStore } from '@reduxjs/toolkit';

import { apiSlice } from 'api/apiSlice';

export const createStore = (cookie?: string) => {
  const preloadedState =
    typeof window !== 'undefined' ? window.__PRELOADED_STATE__ : undefined;

  if (typeof window !== 'undefined') {
    delete window.__PRELOADED_STATE__;
  }

  return configureStore({
    reducer: {
      [apiSlice.reducerPath]: apiSlice.reducer,
    },
    preloadedState,
    middleware: getDefaultMiddleware =>
      getDefaultMiddleware({
        thunk: {
          extraArgument: cookie,
        },
      }).concat(apiSlice.middleware),
  });
};

export type AppDispatch = ReturnType<typeof createStore>['dispatch'];
export type RootState = ReturnType<typeof createStore>['getState'];
