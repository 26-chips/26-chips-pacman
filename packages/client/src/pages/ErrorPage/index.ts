import { lazy } from 'react';

export const ErrorPage = lazy(
  () => /* webpackChunkName: "HeavyComponent" */ import('./ErrorPage')
);
