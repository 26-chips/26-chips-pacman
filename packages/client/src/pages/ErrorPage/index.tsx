import styles from './errorPage.module.scss';
import { Button, Link } from 'components';
import { Images } from 'assets/img';
import { useNavigate } from 'react-router-dom';
import { ROUTES } from 'router';

export interface ErrorPageProps {
  statusCode: string;
  errorText: string;
  image: string;
}

export const ErrorPage = ({
  statusCode = '404',
  errorText = 'Упс!\nКажется, такой страницы не существует',
  image = 'Green',
}: Omit<ErrorPageProps, 'children'>): JSX.Element => {
  const nav = useNavigate();

  return (
    <div className={styles.container}>
      <div className={styles.relativeContainer}>
        <img
          src={Images[image]}
          alt="Error page logo"
          title="Error page logo"
          className={styles.image}
        />

        <h1 className={styles.errorCode}>{statusCode}</h1>

        <p className={styles.pre}>{errorText}</p>
        <Button className={styles.button} onClick={() => nav(-1)}>
          Назад
        </Button>
        <Link to={ROUTES.MAIN}>На главную</Link>
      </div>
    </div>
  );
};
