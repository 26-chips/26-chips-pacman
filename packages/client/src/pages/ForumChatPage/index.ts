import { lazy } from 'react';

export const ForumChatPage = lazy(() => import('./ForumChatPage'));
