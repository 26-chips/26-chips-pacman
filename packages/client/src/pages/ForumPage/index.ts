import { lazy } from 'react';

export const ForumPage = lazy(() => import('./ForumPage'));
