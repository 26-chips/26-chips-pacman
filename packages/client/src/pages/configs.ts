import { REQUIRED } from 'components/Form/validate';

type Password = {
  password: string;
  newPassword: string;
};

export type FieldConfig<T = unknown> = {
  name: string;
  title: string;
  type: string;
  validationType?: 'login' | 'password' | 'email' | 'phone' | 'name';
  inlineTitle?: boolean;
  showDeleteSymbol?: boolean;
  customValidation?: (
    values: T,
    value: string
  ) => 'Поле обязательно для заполнения' | 'Пароли не совпадают' | '';
};

const customValidation = (password: string, value: string) => {
  if (!value) return REQUIRED;
  if (password !== value) return 'Пароли не совпадают';
  return '';
};

export const loginConfig: FieldConfig[] = [
  {
    name: 'login',
    title: 'Логин',
    validationType: 'login',
    type: 'text',
  },
  {
    name: 'password',
    title: 'Пароль',
    validationType: 'password',
    type: 'password',
  },
];

export const profileConfig: FieldConfig[] = [
  {
    name: 'login',
    title: 'Логин',
    validationType: 'login',
    type: 'text',
    inlineTitle: true,
    showDeleteSymbol: false,
  },
  {
    name: 'first_name',
    title: 'Имя',
    type: 'text',
    validationType: 'name',
    inlineTitle: true,
    showDeleteSymbol: false,
  },
  {
    name: 'second_name',
    title: 'Фамилия',
    type: 'text',
    validationType: 'name',
    inlineTitle: true,
    showDeleteSymbol: false,
  },
  {
    name: 'display_name',
    title: 'Никнейм',
    type: 'text',
    validationType: 'name',
    inlineTitle: true,
    showDeleteSymbol: false,
  },
  {
    name: 'email',
    title: 'Почта',
    type: 'email',
    validationType: 'email',
    inlineTitle: true,
    showDeleteSymbol: false,
  },
  {
    name: 'phone',
    title: 'Телефон',
    type: 'tel',
    validationType: 'phone',
    inlineTitle: true,
    showDeleteSymbol: false,
  },
];

export const passwordConfig: FieldConfig<Pick<Password, 'newPassword'>>[] = [
  {
    name: 'oldPassword',
    title: 'Старый пароль',
    type: 'password',
    validationType: 'password',
    inlineTitle: true,
    showDeleteSymbol: false,
  },
  {
    name: 'newPassword',
    title: 'Новый пароль',
    type: 'password',
    validationType: 'password',
    inlineTitle: true,
    showDeleteSymbol: false,
  },
  {
    name: 'repeatPassword',
    title: 'Повторите новый пароль',
    type: 'password',
    inlineTitle: true,
    showDeleteSymbol: false,
    customValidation: ({ newPassword }, value) =>
      customValidation(newPassword, value),
  },
];

export const registrationConfig: FieldConfig<Pick<Password, 'password'>>[] = [
  {
    name: 'email',
    validationType: 'email',
    title: 'Почта',
    type: 'email',
  },
  {
    name: 'first_name',
    validationType: 'name',
    title: 'Имя',
    type: 'text',
  },
  {
    name: 'second_name',
    validationType: 'name',
    title: 'Фамилия',
    type: 'text',
  },
  {
    name: 'phone',
    validationType: 'phone',
    title: 'Телефон',
    type: 'tel',
  },
  {
    name: 'login',
    validationType: 'login',
    title: 'Логин',
    type: 'text',
  },
  {
    name: 'password',
    validationType: 'password',
    title: 'Пароль',
    type: 'password',
  },
  {
    name: 'repeat_password',
    customValidation: ({ password }, value) =>
      customValidation(password, value),
    title: 'Повторите пароль',
    type: 'password',
  },
];
