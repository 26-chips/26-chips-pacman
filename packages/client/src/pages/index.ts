export * from './Layout';
export * from './ErrorPage';
export * from './MainPage';
export * from './GamePage';
export * from './SigninPage';
export * from './SignupPage';
export * from './ProfilePage';
export * from './LeaderboardPage';
export * from './ForumPage';
export * from './ForumChatPage';
export * from './UIPage';
export * from './StartPage';
