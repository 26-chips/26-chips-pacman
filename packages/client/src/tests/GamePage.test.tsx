import { render, act } from '@testing-library/react';
import { GamePage } from 'pages';
import { BrowserRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';
import { store } from '../app/store';
import { Provider } from 'react-redux';

const mockComponent = (
  <Provider store={store}>
    <BrowserRouter>
      <GamePage />
    </BrowserRouter>
  </Provider>
);

describe('<GamePage />', () => {
  it('Should render', async () => {
    //await lazy load https://github.com/testing-library/react-testing-library/issues/1216
    let container;
    await act(() => {
      container = render(mockComponent);
    });
    if (container) {
      // @ts-ignore
      expect(container.getByText(/Время/i)).toBeDefined();
    } else {
      fail('Page not render properly (or not contains "Time" string)');
    }
  });

  it('Should contain canvas', async () => {
    const { container } = await render(mockComponent);
    expect(container.querySelector('canvas')).toBeInTheDocument();
  });

  it('Should match snapshot', () => {
    const tree = create(mockComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
