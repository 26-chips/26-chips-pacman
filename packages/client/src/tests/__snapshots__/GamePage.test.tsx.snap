// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GamePage /> Should match snapshot 1`] = `
Array [
  <div>
    <p>
      Очки:
      <span>
        0
      </span>
    </p>
    <p>
      Оставшиеся жизни:
      <span>
        3
      </span>
    </p>
    <p>
      Время:
      <span>
        0
      </span>
    </p>
  </div>,
  <div>
    <img
      alt="fullscreen switch"
      onClick={[Function]}
      src="default"
    />
  </div>,
  <div
    className=""
  >
    <canvas />
  </div>,
  <div>
    <div
      className=""
    >
      3
    </div>
  </div>,
]
`;
